data/mods/CMakeLists.txt

```diff
@@ -13,6 +20,13 @@ find_package(ZIP REQUIRED)
 set_source_files_properties("${wz2100_autorevision_cache_file}" PROPERTIES GENERATED TRUE)
 set_source_files_properties("${wz2100_autorevision_h_file}" PROPERTIES GENERATED TRUE)
 
+function(_MODINFO_CREATE_BASE_TARGET)
+	if(NOT TARGET mod_info_gen)
+		add_custom_target(mod_info_gen)
+		set_property(TARGET mod_info_gen PROPERTY FOLDER "_WZBuildProcessTargets")
+	endif()
+endfunction()
+
 function(_MODINFO_JSON_GET_UNIQUE_TARGET_NAME _name _unique_name)
```

Comment:
Ideally, there should be a function to parse `mod-info.json`, which will be used to populate the `_unique_name` part of the target name, so that the generated targets have meaningful names.

Right now the relationship between the mod for which we generate the target and the result modinfo target name is completely obscured.

Reply:
For example, we have `string(JSON ...)` in CMake 3.19+, so that we can use it to get modinfo targets with improved readability, and fall back to counter-based approach for older versions (just because parsing JSONs otherwise would be a quite tedious task).